tools/cpp/windows_cc_configure.bzl

```
@@ -254,17 +249,21 @@ def find_vc_path(repository_ctx):
     auto_configure_warning_maybe(repository_ctx, "Visual C++ build tools found at %s" % vc_dir)
     return vc_dir
 
-def _is_vs_2017_or_2019(repository_ctx, vc_path):
-    """Check if the installed VS version is Visual Studio 2017 or 2019."""
+def _is_vs_2017_or_newer(repository_ctx, vc_path):
+    """Check if the installed VS version is Visual Studio 2017 or newer."""
 
-    # The layout of VC folder in VS 2017 and 2019 is different from that in VS 2015 and older versions.
-    # In VS 2017 and 2019, it contains only three directories:
+    # The layout of VC folder in VS 2017 and newer versions is different from that in VS 2015 and older versions.
+    # From VS 2017 it contains three directories:
     # "Auxiliary", "Redist", "Tools"
+    # From VS 2022 17.6, a fourth "vcpkg" directory is also present
 
-    vc_2017_or_2019_contents = ["auxiliary", "redist", "tools"]
+    vc_2017_or_newer_contents = ["auxiliary", "redist", "tools"]
     vc_path_contents = [d.basename.lower() for d in repository_ctx.path(vc_path).readdir()]
```

Maybe a different approach would be to future-proof this check:
```suggestion
    vc_path_contents = [d.basename.lower() for d in repository_ctx.path(vc_path).readdir() if d.basename.lower() in vc_2017_or_2019_contents]
```

---

I was thinking about it, although I did not have a 2015 VS installation to rule out that it does not contain those entries plus something else. But that could be a way to future-proof this check, yes.

---

done (I've double checked this will not break down on a 2015 VC)